### ISSUE TITLE
Device Token Null, Refresh Never Called

### ISSUE THREAD

**Issue by jeremy4772** [evollu/react-native-fcm#489]
Please include following information for better support

What version of RN and react-native-fcm are you running?

React Native version 0.47.1, fcm version 7.5.1

What device are you using? (e.g iOS9 emulator, Android 6 device)?

Android 7 device, installed with react-native run-android

Is your app running in foreground, background or not running?

Foreground

I've been stepping through the README trying to integrate this into an existing application (which is production so can't share code). I did made the google-secrets.json file, installed and linked fcm, and modified all of the gradle and manifest files as specified in the README, but when I try to do

`    FCM.requestPermissions(); // for iOS

    FCM.getFCMToken().then(token => {
      console.log("Get Token " + token);
      // store fcm token in your server
    });

    this.refreshTokenListener = FCM.on(FCMEvent.RefreshToken, (token) => {
      console.log("Refresh Token " + token);

      // fcm token may not be available on first load, catch it here
    });`

The getToken always gives me null and the refresh token listener never fires at all.

Any advice? I'm honestly not sure if this is a bug or if I'm just doing something wrong.

**Comment by jeremy4772** [comment]
Edit: Resolved by running gradlew clean, removing and reinstalling node_modules, and the re running react-native run-android

**Comment by EroticCode** [comment]
@jeremy4772  my app not working please help

**Comment by jeremy4772** [comment]
After I did that it worked exactly as planned. Although since the time of this post, my team has moved onto https://github.com/zo0r/react-native-push-notification (the library worked fine, but we have different needs than three weeks ago).

In detail exactly what I did was (from your main app directory):

`rm -rf node_modeules`

`npm install`

`cd android`

`gradlew clean`

and after that it was fine. Are you on a real device? You will never get a valid push token on any emulator (iOS or Android) regardless of what library you're using.
